packages/react/radio-group/src/RadioGroup.tsx

```diff
@@ -146,7 +165,7 @@ const RadioGroupItem = React.forwardRef<RadioGroupItemElement, RadioGroupItemPro
              * We do this imperatively instead of updating `context.value` because changing via
              * state would not trigger change events (e.g. when in a form).
              */
```

Comment:
I wonder if we need to fledge this comment to explain why we do the arrow check?

Reply:
Does this line cover it? https://github.com/radix-ui/primitives/pull/939/files#diff-39138edb5652b34ca612c34d47986e2e8eb2c97315f178b7fe237b28e0a2aeeaR164

I thought it implied arrow keypress since that's what roving focus does but let me know if you'd prefer something more explicit here.

Reply:
Sorry I can't see where this is pointing to for some reason…

Reply:
Ah it's because GitHub collapses that line ...
![image](https://user-images.githubusercontent.com/175330/141126524-b4d2cffa-3aa3-455a-a09c-01101e3defb8.png)

Reply:
To be honest the second half of that comment confuses me now 😀
The first half is correct, although I feel like it could be clearer now that we've added the arrow keys check.

Perhaps something like:

```
Our `RovingFocusGroup` will focus item as we navigate with arrow keys.
We need to make that "check" that item when that's the case.
However, programatic focus should not "check" the focused item.
```

Reply:
The second part of the comment is explaining why we imperatively call `.click()` to check it instead of setting `context.value` to the current items value (declarative). I think we still need that information.

Does this work?

```jsx
Our `RovingFocusGroup` will focus the radio when navigating with arrow keys and 
we need to "check" it in that case. We click it to "check" it (instead of updating 
`context.value`) so that the radio change event fires.
```